crates/cubecl-wgpu/src/runtime.rs

```diff
@@ -90,95 +82,112 @@ impl Default for RuntimeOptions {
     }
 }
 
-pub fn init_existing_device(
-    adapter: Arc<wgpu::Adapter>,
-    device: Arc<wgpu::Device>,
-    queue: Arc<wgpu::Queue>,
-    options: RuntimeOptions,
-) -> WgpuDevice {
-    let device_id = WgpuDevice::Existing(device.as_ref().global_id());
-    let client = create_client(adapter, device, queue, options);
+/// A complete setup used to run wgpu on a GPU.
+///
+/// These can either be created with [`ini
+#[derive(Clone)]
+pub struct WgpuSetup {
+    /// The underlying wgpu instance.
+    pub instance: Arc<wgpu::Instance>,
+    /// The chose 'adapter'. This corresponds to a physical device.
+    pub adapter: Arc<wgpu::Adapter>,
+    /// The wpgu device Burn will use. Nb: There can only be one device per adapter.
+    pub device: Arc<wgpu::Device>,
+    /// The queue Burn commands will be submittd to.
+    pub queue: Arc<wgpu::Queue>,
+}
+
+/// Create a `WgpuDevice` on an existing `WgpuSetup`. Useful when you want to share
+/// a device between CubeCL and other wgpu libraries.
+pub fn init_device_on_setup(setup: WgpuSetup, options: RuntimeOptions) -> WgpuDevice {
```

Comment:
They are confusing, are they all creating and registering devices?

- `fn init_device_on_setup(WgpuSetup) -> WgpuDevice`
- `async fn init_device(WgpuDevice) -> WgpuSetup`
- `fn init_device_async(WgpuDevice) -> WgpuSetup`

Reply:
Init_device and init_device_async are for initializing a cube WgpuDevice, and create all the wgpu resources.

init_device_on_setup initializes a cube WgpuDevice given existing wgpu resources.

Sorry if they are a bit confusing, maybe other names could be better!

Reply:
@ArthurBrussee I have figured out their usage. However, I revised function names based on suggestions from @nathanielsimard in #214.